src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1157,12 +1198,20 @@ ApplicableJoinClauses(List *leftTableIdList, uint32 rightTableId, List *joinClau
 }
 
 
+/* Returns the left expression in the given join clause. */
+Node *
+LeftHandExpr(OpExpr *joinClause)
```

Comment:
This function can be static (and moved after it's first usage). (same for `RightHandExpr`)

Reply:
Now it could be static yes, but I hope we can soon fix the partition joins to allow for repartitions on expressions instead of `Var`'s which would need this function again.

That said, `LeftColumn` and `RightColumn` could probably be removed as you stated below. That would make these lines prime for inlining in the `...OrNULL` variants of both, causing these functions to be removed.